### PR TITLE
Update Keptn extension to V0.8.2

### PR DESCRIPTION
**Pull request by TannerGabriel**
Updating the Keptn marketplace extension to the newest version (v0.8.2) as discussed with @jetzlstorfer.

* [X] Notified the Maintainer of the application in this pull request so that they are aware of your proposal.
* [X] Outlined the changes you are proposing, and the reasons these are required (e.g. stability, compatibility with new versions of Kubernetes implementations, etc).
* [ ] Tested that the application works with the proposed updates applied (including a screenshot)

### CODE REVIEW

**Comment by jetzlstorfer** [comment]
Just tried with the new instructions but installation fails.

```
Events:
  Type     Reason              Age                  From                     Message
  ----     ------              ----                 ----                     -------
  Warning  FailedScheduling    3m23s                default-scheduler        0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims.
  Warning  FailedScheduling    3m23s                default-scheduler        0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims.
  Normal   Scheduled           2m38s                default-scheduler        Successfully assigned keptn/configuration-service-54c4d66bbb-ppjgv to k3s-civo-marketplace-638b6938-node-pool-a4a9
  Warning  FailedAttachVolume  79s (x2 over 112s)   attachdetach-controller  AttachVolume.Attach failed for volume "pvc-3d8c3b7d-a521-40d1-9312-8a394cd95fea" : rpc error: code = Unknown desc = UnknownError
  Warning  FailedMount         36s                  kubelet                  Unable to attach or mount volumes: unmounted volumes=[configuration-volume], unattached volumes=[configuration-volume keptn-configuration-service-token-wpxhs]: timed out waiting for the condition
  Warning  FailedAttachVolume  15s (x7 over 2m24s)  attachdetach-controller  AttachVolume.Attach failed for volume "pvc-3d8c3b7d-a521-40d1-9312-8a394cd95fea" : rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

@saiyam1814 are PVCs supported by Civo? Seems we are having issues with it if I read the error message correctly.

**Comment by jetzlstorfer** [comment]
Just tried with the new instructions but installation fails.

```
Events:
  Type     Reason              Age                  From                     Message
  ----     ------              ----                 ----                     -------
  Warning  FailedScheduling    3m23s                default-scheduler        0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims.
  Warning  FailedScheduling    3m23s                default-scheduler        0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims.
  Normal   Scheduled           2m38s                default-scheduler        Successfully assigned keptn/configuration-service-54c4d66bbb-ppjgv to k3s-civo-marketplace-638b6938-node-pool-a4a9
  Warning  FailedAttachVolume  79s (x2 over 112s)   attachdetach-controller  AttachVolume.Attach failed for volume "pvc-3d8c3b7d-a521-40d1-9312-8a394cd95fea" : rpc error: code = Unknown desc = UnknownError
  Warning  FailedMount         36s                  kubelet                  Unable to attach or mount volumes: unmounted volumes=[configuration-volume], unattached volumes=[configuration-volume keptn-configuration-service-token-wpxhs]: timed out waiting for the condition
  Warning  FailedAttachVolume  15s (x7 over 2m24s)  attachdetach-controller  AttachVolume.Attach failed for volume "pvc-3d8c3b7d-a521-40d1-9312-8a394cd95fea" : rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

@saiyam1814 are PVCs supported by Civo? Seems we are having issues with it if I read the error message correctly.

**Comment by saiyam1814** [comment]
civo Volume issue is resolved now, can you test one more time and see if it works ? @jetzlstorfer

**Comment by saiyam1814** [comment]
IT took 5 minutes to get this running but as soon as it all went running I saw error 
Error: timed out waiting for the condition -> can we increase the timeout ? 

```
NAME                                     READY   STATUS    RESTARTS   AGE
svclb-api-gateway-nginx-vcb7q            1/1     Running   0          5m26s
svclb-api-gateway-nginx-xscrg            1/1     Running   0          5m26s
svclb-api-gateway-nginx-6r629            1/1     Running   0          5m26s
api-gateway-nginx-79b6bb7767-lhd7d       1/1     Running   0          5m25s
api-service-dc8f9d9dd-k4wmg              2/2     Running   0          5m24s
secret-service-7ffdf597d7-sf674          1/1     Running   0          5m24s
keptn-nats-cluster-0                     3/3     Running   0          5m24s
bridge-597967d475-pftvr                  1/1     Running   0          5m24s
mongodb-datastore-84c975559-v95pp        2/2     Running   2          5m25s
statistics-service-7bd9968dbd-qbjdk      2/2     Running   2          5m24s
remediation-service-5b9ffb54f9-8cr6g     2/2     Running   2          5m24s
shipyard-controller-6844695949-x2m2s     2/2     Running   2          5m25s
lighthouse-service-857df5646b-66gst      2/2     Running   2          5m25s
approval-service-dcb7979c8-d8chr         2/2     Running   2          5m24s
configuration-service-54c4d66bbb-xt4kr   1/1     Running   0          5m25s
mongodb-6b455c56fb-8tw6m                 1/1     Running   0          5m24s
```

**Comment by jetzlstorfer** [comment]
Thanks @saiyam1814 for the update!

I've increased waiting time to 10m and will update this in the PR.

I tried again and it is now running. **Important details** Traefik has to be disabled and should not be installed as it takes the free ports that are also needed to expose keptn via the LB.

Running installation:
```
NAME                                     READY   STATUS    RESTARTS   AGE
svclb-api-gateway-nginx-qtl4c            1/1     Running   0          18m
api-gateway-nginx-f454c64c4-dq9tv        1/1     Running   0          18m
bridge-597967d475-vc428                  1/1     Running   0          18m
secret-service-7ffdf597d7-2rmkt          1/1     Running   0          18m
api-service-dc8f9d9dd-54g7k              2/2     Running   0          18m
keptn-nats-cluster-0                     3/3     Running   0          18m
mongodb-datastore-84c975559-6hpcb        2/2     Running   2          18m
remediation-service-5b9ffb54f9-mbdrd     2/2     Running   2          18m
approval-service-dcb7979c8-6sf74         2/2     Running   2          18m
shipyard-controller-6844695949-f4shv     2/2     Running   2          18m
lighthouse-service-857df5646b-nqvl5      2/2     Running   2          18m
statistics-service-7bd9968dbd-grtx8      2/2     Running   2          18m
configuration-service-54c4d66bbb-4j8m4   1/1     Running   0          18m
mongodb-6b455c56fb-gqjcg                 1/1     Running   0          18m
```

**Comment by saiyam1814** [comment]
Yeah There are some apps that do not work with Traefik, it's ok in future we will try to come up with a mechanism to detect an ingress controller running and say warning to uninstall the previous one or something like that. Apart from this is PR complete now ?

File: keptn/install.sh [hunk 1]
```diff
@@ -1,8 +1,3 @@
 #!/bin/bash
 
-curl -sL https://get.keptn.sh | sudo -E bash
-
-echo "{}" > creds.json
-
-keptn install --endpoint-service-type=NodePort -c ./creds.json
-
+helm upgrade keptn keptn --install -n keptn --create-namespace --wait --version=0.8.2 --repo=https://storage.googleapis.com/keptn-installer --set=control-plane.apiGatewayNginx.type=LoadBalancer
```
Comment (jetzlstorfer) [inline keptn/install.sh:3]: Please update to 
```
helm upgrade keptn keptn --install -n keptn --create-namespace --wait --version=0.8.3 --repo=https://storage.googleapis.com/keptn-installer --set=control-plane.apiGatewayNginx.type=LoadBalancer --timeout 10m
```

File: keptn/manifest.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 name: keptn
-version: 0.7.1
+version: 0.8.2
```
Comment (jetzlstorfer) [inline keptn/manifest.yaml:3]: 0.8.3

File: keptn/post_install.md [hunk 3]
```diff
@@ -18,18 +18,16 @@
...
+  KEPTN_ENDPOINT=http://$(kubectl -n keptn get service api-gateway-nginx -o jsonpath='{$.status.loadBalancer.ingress[0].ip}')/api
   KEPTN_API_TOKEN=$(kubectl get secret keptn-api-token -n keptn -ojsonpath={.data.keptn-api-token} | base64 --decode)
 
   keptn auth --endpoint=$KEPTN_ENDPOINT --api-token=$KEPTN_API_TOKEN
   ```
 
 1. Access the bridge via:
   ```
-  echo http://${EXTERNAL_NODE_IP}:${API_PORT}/bridge
+  echo http://$(kubectl -n keptn get service api-gateway-nginx -o jsonpath='{$.status.loadBalancer.ingress[0].ip}')/bridge
```
Comment (jetzlstorfer) [inline keptn/post_install.md:30]: Please merge the instruction for accessing the bridge.
The following command will give you the endpoint + credentials.
```
keptn configure bridge --output
```

**Review by jetzlstorfer — CHANGES_REQUESTED** [review]

**Comment by jetzlstorfer** [comment]
I'll leave it to the original author of the PR to update some minor issues.
@TannerGabriel please have a look

Once this is addressed we can move it out of "Draft" and ask for the PR to be merged

File: keptn/manifest.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
 maintainer: sangambiradar@hotmail.com
```
[the remark this answers is not in the document]
Comment (saiyam1814) [reply keptn/manifest.yaml:4]: YEs this is important

**Comment by jetzlstorfer** [comment]
@saiyam1814 it is ready for review

File: keptn/manifest.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ---
 name: keptn
-version: 0.7.1
-maintainer: sangambiradar@hotmail.com
-description: an event-based control plane for continuous delivery and automated operations for cloud-native applications.
+version: 0.8.3
+maintainer: gabriel.tanner@dynatrace.com, sangambiradar@hotmail.com
+description: 'Keptn is an event-based control plane for continuous delivery and automated operations for cloud-native applications. Please note: Traefik must not be selected since Keptn uses Istio as an Ingress Controller.'
```
Comment (DoNnMyTh) [inline keptn/manifest.yaml:5]: Please shorten up the description to 160 chars

**Review by DoNnMyTh — CHANGES_REQUESTED** [review]
@jetzlstorfer Please make the requested change

**Review by DoNnMyTh — APPROVED** [review]
can be merged

File: keptn/manifest.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 ---
 name: keptn
-version: 0.7.1
-maintainer: sangambiradar@hotmail.com
-description: an event-based control plane for continuous delivery and automated operations for cloud-native applications.
+version: 0.8.3
```
Comment (saiyam1814) [inline keptn/manifest.yaml:3]: version should be a string @TannerGabriel

Comment (TannerGabriel) [reply keptn/manifest.yaml:3]: Sorry thanks for updating @saiyam1814.

**Pull request merged**